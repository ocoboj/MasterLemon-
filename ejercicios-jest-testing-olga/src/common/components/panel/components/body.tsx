import * as React from 'react';

export const Body: React.StatelessComponent = (props) => (
  <ul className="list-group list-group-flush">
    <li className="list-group-item">
      {props.children}
    </li>
  </ul>
);
