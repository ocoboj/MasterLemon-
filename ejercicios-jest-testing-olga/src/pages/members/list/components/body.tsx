import * as React from 'react';
import { Member } from '../viewModel';
import { Row } from './row';

interface Props {
  members: Member[];
}

export const Body: React.StatelessComponent<Props> = (props) => (
  <tbody>
    {
      props.members.map((member) => (
        <Row
          key={member.id}
          member={member}
        />
      ))
    }
  </tbody>
);
