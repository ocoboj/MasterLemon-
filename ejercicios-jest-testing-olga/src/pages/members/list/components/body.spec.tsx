import * as React from 'react';
import { Body } from './body';

import { mount } from 'enzyme';


describe('pages/members/list/components/body specs', () => {
  it('should render as expected when passing one item', () => {

    const props ={
      members:[
          {
              id: 5,
              name: 'name test',
              avatarUrl: 'url test',
          }
        ]    
    }

    const component = mount(<Body {...props}/>);

    expect(component).toMatchSnapshot();
  });

  it('should render as expected when passing more than one item ', () => {

    const props = {
      members:[
        {
          id: 6,
          name: 'name test 6',
          avatarUrl: 'url test 6',
        } ,
        {
          id: 8,
          name: 'name test 8',
          avatarUrl: 'url test 8',
       } 
      ]
    };

    const component = mount(<Body {...props}/>);

    expect(component).toMatchSnapshot();
   
  });



});