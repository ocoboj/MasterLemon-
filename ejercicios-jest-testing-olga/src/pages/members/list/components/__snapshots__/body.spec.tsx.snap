// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/body specs should render as expected when passing more than one item  1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test 6",
        "id": 6,
        "name": "name test 6",
      },
      Object {
        "avatarUrl": "url test 8",
        "id": 8,
        "name": "name test 8",
      },
    ]
  }
>
  <tbody>
    <Component
      key="6"
      member={
        Object {
          "avatarUrl": "url test 6",
          "id": 6,
          "name": "name test 6",
        }
      }
    >
      <tr>
        <td>
          <img
            src="url test 6"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            6
          </span>
        </td>
        <td>
          <span>
            name test 6
          </span>
        </td>
      </tr>
    </Component>
    <Component
      key="8"
      member={
        Object {
          "avatarUrl": "url test 8",
          "id": 8,
          "name": "name test 8",
        }
      }
    >
      <tr>
        <td>
          <img
            src="url test 8"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            8
          </span>
        </td>
        <td>
          <span>
            name test 8
          </span>
        </td>
      </tr>
    </Component>
  </tbody>
</Component>
`;

exports[`pages/members/list/components/body specs should render as expected when passing one item 1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test",
        "id": 5,
        "name": "name test",
      },
    ]
  }
>
  <tbody>
    <Component
      key="5"
      member={
        Object {
          "avatarUrl": "url test",
          "id": 5,
          "name": "name test",
        }
      }
    >
      <tr>
        <td>
          <img
            src="url test"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            5
          </span>
        </td>
        <td>
          <span>
            name test
          </span>
        </td>
      </tr>
    </Component>
  </tbody>
</Component>
`;
